scvi/data/anndata/fields/_obsm_field.py

```diff
@@ -92,11 +97,100 @@ def transfer_field(
         **kwargs,
     ) -> None:
         super().transfer_field(setup_dict, adata_target, **kwargs)
-        self.validate_field(adata_target)
         self.register_field(adata_target)
 
     def compute_summary_stats(self, adata: AnnData) -> dict:
         super().compute_summary_stats(adata)
         n_continuous_covariates = len(self.obs_keys)
         stat_name = f"n_{self.registry_key}"
         return {stat_name: n_continuous_covariates}
+
+
+class CategoricalJointObsField(JointObsField):
+    """
+    An AnnDataField for a collection of categorical .obs fields in the AnnData data structure.
+
+    Parameters
+    ----------
+    registry_key
+        Key to register field under in data registry.
+    obs_keys
+        Sequence of keys to combine to form the obsm field.
+    """
+
+    def __init__(self, registry_key: str, obs_keys: Optional[List[str]]) -> None:
+        super().__init__(registry_key, obs_keys)
+        self._mappings_key = registry_key
+
+    def _initialize_mappings_dict(self, adata: AnnData) -> None:
+        adata.uns[_constants._SETUP_DICT_KEY][self._mappings_key] = dict(
+            mappings=dict(), keys=[]
+        )
+
+    def _make_obsm_categorical(
```

Comment:
does this make obsm categorical? It looks like it just records the categories per covariate

Reply:
good point

Reply:
should we change the name?

Reply:
i made it so it does make it categorical